dotnet/src/SemanticKernel.Abstractions/AI/TextCompletion/ITextCompletion.cs

```diff
@@ -20,7 +21,7 @@ public interface ITextCompletion
     /// <returns>Text generated by the remote model</returns>
     public Task<string> CompleteAsync(
         string text,
-        CompleteRequestSettings requestSettings,
+        JsonObject requestSettings,
```

Comment:
we looked into this approach 2-3 months ago and agreed not to use generic object/dictionaries, and use type safety, clearly defined types and interfaces

Reply:
Assuming the settings should be flexible, I'm not sure it will be possible to define types to cover all cases, AI types and models.